### PR TITLE
Mapgen.lua: Various fixes and improvements

### PR DESCRIPTION
**Pull request by paramat**
Nodes:
Add savanna nodes (appear in non-mgv6 mapgens only):
Add 'dirt with dry grass'.
Add 'dry grass' (dry tall grass).
Textures were derived from the green grass textures.

Mapgen aliases:
desert_stone does not exist, use redsandstone.
desert_sand does not exist, use redsand.
sandstonebrick does not exist, use sandstone.
stair_sandstonebrick does not exist, use stair_sandstone.

Ores:
Generate bedrock in lava_source too.
Change desert_stone to redsandstone.

Biomes:
Change desert_stone to redsandstone.
Change desert_sand to redsand.

Decorations:
Make bonusbox chest appear on all mgv6 surface nodes.
Add bonusbox chest to non-mgv6 mapgens.
Replace redundant multiple grass decorations with an equivalent single grass decoration,
because only 1 length is used in MultiCraft.
Change desert_sand to redsand.

General format cleanup.
///////////////////

screenshot_20180720_023109

^ Mgv6 desert now has redsandstone, redsand, cacti and dry shrubs

screenshot_20180720_023759

^ Mgv7p desert now has correct node for dungeon walls

screenshot_20180720_024254

^ Mgv7p desert now has redsandstone, redsand, cacti and dry shrubs

screenshot_20180720_024342

^ Sandstone biome dungeons now have correct node for dungeon walls

screenshot_20180720_025424

^ Bedrock now cuts through lava caves. It's still not completely continuous but better.

screenshot_20180726_024103

screenshot_20180726_024415

^ Savanna with new nodes.

### CODE REVIEW

**Comment by MoNTE48** [comment]
@paramat please, write me email, when you are online. Thanks

**Comment by paramat** [comment]
Updated.
The new savanna textures were derived from the green grass textures.
Also added the bonus box decorations to non-mgv6 mapgens.

File: games/default/files/default/nodes.lua [hunk 1]
```diff
@@ -1710,21 +1723,6 @@ minetest.register_node("default:grass", {
     wield_image = "default_tallgrass.png",
     walkable = false,
     buildable_to = true,
-        --drop = {
```
Comment (MoNTE48) [inline games/default/files/default/nodes.lua]: Please, back this!
I uncomment this code, when farming mod will be done.

Comment (paramat) [reply games/default/files/default/nodes.lua]: Done.

**Comment by MoNTE48** [comment]
@paramat The most important question.
Existing worlds mapgen v6 will not have problems with unknown nodes?

**Comment by paramat** [comment]
I'm not sure, so don't merge yet.
I'll test to see.

**Comment by paramat** [comment]
Tested a mgv6 moving from master branch to PR branch.
No world breakage, no unknown nodes.

**Comment by paramat** [comment]
@MoNTE48 that took about 6 hrs. work in total :)

**Comment by MoNTE48** [comment]
@paramat please, reply on my email!

**Comment by paramat** [comment]
I did.

**Comment by paramat** [comment]
Ok just seen your new email.

**Comment by XREALM-YT** [comment]
> @paramat please, write me email, when you are online. Thanks

@MoNTE48 can i give update ideas? Or not

**Pull request merged**